android/java/org/chromium/chrome/browser/settings/BraveSyncScreensPreference.java

```diff
@@ -719,9 +751,101 @@ public void onTextChanged(CharSequence s, int start, int before, int count) {
             setNewChainLayout();
         } else if (mDeleteAccountButton == v) {
             permanentlyDeleteAccount();
+        } else if (mNewCodeWordsButton == v) {
+            generateNewCodeWords();
+        } else if (mNewQrCodeButton == v) {
+            generateNewQrCode();
         }
     }
 
+    void generateNewCodeWords() {
+        getActivity()
+                .runOnUiThread(
+                        new Runnable() {
+                            @Override
+                            public void run() {
+                                String codePhrase = getPureWords();
+                                assert codePhrase != null && !codePhrase.isEmpty();
+                                String timeLimitedWords =
+                                        getBraveSyncWorker()
+                                                .getTimeLimitedWordsFromPure(codePhrase);
+                                assert timeLimitedWords != null && !timeLimitedWords.isEmpty();
+                                mBraveSyncAddDeviceCodeWords.setVisibility(View.VISIBLE);
+                                mBraveSyncAddDeviceCodeWords.setText(timeLimitedWords);
+                                mCopyButton.setVisibility(View.VISIBLE);
+                                mNewCodeWordsButton.setVisibility(View.GONE);
+
+                                LocalDateTime notAfterTime =
+                                        getBraveSyncWorker()
+                                                .getNotAfterFromFromTimeLimitedWords(
+                                                        timeLimitedWords);
```

Comment:
the same as the above

Reply:
the same, https://github.com/brave/brave-core/commit/43cafddcff2013f279118eb258965f7406b4b3cb